src/objects/zif_abapgit_rule.intf.abap

```diff
@@ -0,0 +1,37 @@
+INTERFACE zif_abapgit_rule
+  PUBLIC .
+  TYPES ty_fill_rule TYPE c LENGTH 2.
+  CONSTANTS:
+    BEGIN OF c_fill_rule,
+      date      TYPE ty_fill_rule VALUE 'DT',
+      time      TYPE ty_fill_rule VALUE 'TM',
+      timestamp TYPE ty_fill_rule VALUE 'TS',
+      user      TYPE ty_fill_rule VALUE 'UR',
+      client    TYPE ty_fill_rule VALUE 'CT',
+    END OF c_fill_rule.
+  TYPES:
+    BEGIN OF ty_item,
+      tabname     TYPE tabname,
+      fieldname   TYPE fieldname,
+      clear_field TYPE abap_bool,
+      fill_rule   TYPE ty_fill_rule,
+    END OF ty_item,
+    ty_items TYPE SORTED TABLE OF ty_item WITH UNIQUE KEY tabname fieldname.
+
+  METHODS add_item
```

Comment:
I suggest to change this to `add` with parameters `iv_table`, `iv_field`, `iv_fill_rule`. also use `returning self` which will allow to chain the rule definitions (see https://github.com/abapGit/abapGit/blob/main/src/ui/zcl_abapgit_gui_page_sett_locl.clas.abap#L116 for an example in another context).

Reply:
Replaced with add method along with individual parameters. I have been following the clean code approach and would like to know the details of why individual parameters are recommended. It is just for my knowledge. Appreciate your reply on this.

Reply:
It simplifies the calling code a lot since you don't need to fill local variables/itabs to call the method.